provisioning/transport/http/devdoc/http_requirements.md

```diff
@@ -0,0 +1,129 @@
+  api-version: string;
+  operationStatusPollingInterval: number
+  constructor(idScope: string, registrationId: string) : void;
+  register(authorization: string, forceRegistration: boolean, callback: (err: Error, result?: Any) => void) : void;
+  disconnect(callback: (err) => void);
+}
+```
+
+## Example Usage
+```js
+'use strict';
+var Http = require('azure-iot-provisioning-http').Http;
+
+var http = new Http('__MYSCOPE__', '__MY_DEVICE_ID__');
+
+http.register(undefined, false, function(err, body)) {
+  if (err) {
+    if (err instanceof 'UnauthorizedError') {
+      console.log('Unauthorized: ' + body.message); // expect to see 'Authorization required, resend request using supplied key'
+    } else {
+      console.log('unknown error: ' + err.toString());
+    }
+  } else {
+    console.log('Success: status = ' + body.status);
+  }
+}
+
+
+
+```
+
+## Public Interface
+
+### constructor(idScope: string, registrationId: string) : void
+The `constructor` creates an Http transport object used to communicate with the Azure device provisioning service
+
+**SRS_NODE_PROVISIONING_HTTP_18_001: [** The `Http` constructor shall accept the following properties:
+- `idScope` - a string specifiying the scope of the provisioning operations,
+- `registrationId` - the registration id for the specific device 
+**]**
+
+
+### register(authorization: string, forceRegistration: boolean, callback: (err: Error, result?: Any) => void) : void;
+`register` calls into the service to register the given device with the provisioning service.
+
+**SRS_NODE_PROVISIONING_HTTP_18_002: [** `register` shall throw a `ReferenceError` if `callback` is falsy. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_036: [** `register` shall call the `callback` with an `InvalidOperationError` if it is called while a previous registration is in progress. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_004: [** The registration request shall use Http 1.1. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_005: [** The registration request shall include the current `api-version` as a URL query string value named 'api-version'. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_006: [** The registration request shall specify the following in the Http header:
+  Accept: application/json
+  Content-Type: application/json; charset=utf-8 **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_007: [** If an `authorization` string is specifed, it shall be URL encoded and included in the Http Authorization header. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_008: [** If `forceRegistration` is specified, the registration request shall include this as a query string value named 'forceRegistration' **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_009: [** `register` shall POST the registration request to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}' **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_010: [** `register` shall wait for the response to the POST request. **]**
```

Comment:
>[** `register` shall wait for the response to the POST request. **]** [](start = 37, length = 69)

so the idea is that register calls the callback after a successful POST request and then we use events to notify the user of progress, am I getting this right?  can the example usage reflect that?